test/util/utils.js

```diff
@@ -0,0 +1,150 @@
+ * @returns {string} clr command
+ */
+exports.clrCmd = function clrCmd(cmd) {
+  return exports.isWindows ? cmd : ('mono ' + cmd);
+};
+
+/**
+ * Gets the name of the current branch to which the PR is sent.
+ * @returns {string} branchName The current branch name.
+ */
+exports.getCurrentBranch = function getCurrentBranch() {
+  let cmd = 'git rev-parse --abbrev-ref HEAD';
+  let result = 'master';
+  try {
+    result = execSync(cmd, { encoding: 'utf8' });
+  } catch (err) {
+    console.log(`An error occurred while getting the current branch ${util.inspect(err, { depth: null })}.`);
+  }
+  console.log(`>>>>> The current branch is: "${result}".`);
+  return result;
+};
+
+/**
+ * Retrieves list of swagger files to be processed for linting
+ * @returns {Array} list of files to be processed for linting
+ */
+exports.getFilesChangedInPR = function getFilesChangedInPR() {
+  let result = exports.swaggers;
+  if (exports.prOnly === 'true') {
+    let currentBranch, cmd, filesChanged, swaggerFilesInPR;
+    try {
+      currentBranch = exports.getCurrentBranch();
+      cmd = `git diff --name-only HEAD $(git merge-base HEAD ${currentBranch})`;
+      filesChanged = execSync(cmd, { encoding: 'utf8' });
+      console.log('>>>>> Files changed in this PR are as follows:');
+      console.log(filesChanged);
+      swaggerFilesInPR = filesChanged.split('\n').filter(function (item) {
+        return (item.match(/.*\/swagger\/*/ig) !== null);
+      });
+      console.log(`>>>> Number of swaggers found in this PR: ${swaggerFilesInPR.length}`);
+      result = swaggerFilesInPR;
+    } catch (err) {
+      throw err;
+    }
+  }
+  return result;
+};
+
+/**
+ * Downloads the remote schemas and initializes the validator with remote references.
+ * @returns {Object} context Provides the schemas in json format and the validator.
+ */
+exports.initializeValidator = function initializeValidator(callback) {
+  request({ url: exports.extensionSwaggerSchemaUrl, json: true }, function (error, response, extensionSwaggerSchemaBody) {
+    if (error) {
+      return callback(error);
+    }
+    request({ url: exports.swaggerSchemaAltUrl, json: true }, function (error, response, swaggerSchemaBody) {
+      if (error) {
+        return callback(error);
+      }
+      request({ url: exports.exampleSchemaUrl, json: true }, function (error, response, exampleSchemaBody) {
+        if (error) {
+          return callback(error);
+        }
+        request({ url: exports.compositeSchemaUrl, json: true }, function (error, response, compositeSchemaBody) {
+          if (error) {
+            return callback(error);
+          }
+          let context = {
+            extensionSwaggerSchema: extensionSwaggerSchemaBody,
+            swaggerSchema: swaggerSchemaBody,
+            exampleSchema: exampleSchemaBody,
+            compositeSchema: compositeSchemaBody
+          };
+          let validator = new z({ breakOnFirstError: false });
```

Comment:
z is the z-schema package that we depend on for validating a json document against a given schema.

Top of the file should have a variable named z defined.